lib/chef/file_content_management/tempfile.rb

```diff
@@ -63,12 +64,18 @@ def tempfile_open
       # as the arguments to Tempfile.new() consistently.
       #
       def tempfile_basename
-        basename = ::File.basename(@new_resource.name)
+        basename = ::File.basename(@new_resource.path, tempfile_extension)
+        # the leading "[.]chef-" here should be considered a public API and should not be changed
```

Comment:
that strips it so we get the prefixey part of the basename so that "foo.html.erb" becomes "chef-foo" here and the extension becomes ".html.erb"